### PR TITLE
Create{Container,Pod}SecurityContext: modify a pod and don't return the annotations

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it**:
Prior https://github.com/kubernetes/kubernetes/pull/52849 we couldn't modify a pod and had to return annotations from the methods. But now, as we always working with a copy of a pod, we can modify it directly and we don't need to copy&return annotations separately.

This PR simplifies the code by modifying a pod directly. Also it renames these methods and replaces returning of the `SecurityContext` by in-place modification.

In fact it reverts the changes from https://github.com/kubernetes/kubernetes/pull/30257

**Release note**:
```release-note
NONE
```

PTAL @liggitt @timstclair 
CC @simo5

### CODE REVIEW

**Comment by liggitt** [comment]
/ok-to-test
LGTM, will let @tallclair look as well

File: pkg/security/podsecuritypolicy/types.go [hunk 1]
```diff
@@ -32,12 +32,10 @@ import (
...
 type Provider interface {
-	// Create a PodSecurityContext based on the given constraints. Also returns an updated set
-	// of Pod annotations for alpha feature support.
-	CreatePodSecurityContext(pod *api.Pod) (*api.PodSecurityContext, map[string]string, error)
-	// Create a container SecurityContext based on the given constraints. Also returns an updated set
-	// of Pod annotations for alpha feature support.
-	CreateContainerSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, map[string]string, error)
+	// Create a PodSecurityContext based on the given constraints.
+	CreatePodSecurityContext(pod *api.Pod) (*api.PodSecurityContext, error)
+	// Create a container SecurityContext based on the given constraints.
```
Comment (tallclair) [inline pkg/security/podsecuritypolicy/types.go]: nit: add a note that the pod is modified.

It's a little confusing to have the pod be modified, but also return a context that is then expected to be added to the pod... perhaps this API should just be changed to modify the pod directly?

Comment (liggitt) [reply pkg/security/podsecuritypolicy/types.go]: Even returning the security context is no longer necessary. I agree… the create... functions should be renamed to default…, add modify the pod and container directly, but a follow up  to do that is OK with me

Comment (php-coder) [reply pkg/security/podsecuritypolicy/types.go]: Thanks for the feedback! Updated. PTAL.

**Comment by php-coder** [comment]
@tallclair @liggitt PTAL

**Comment by liggitt** [comment]
nice cleanup, is fine for 1.10
/lgtm

**Comment by php-coder** [comment]
> /approve

@brendandburns Could you (or someone else) approve it with `/approve no-issue`? Thanks!

**Pull request merged**